scripts/sign.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+set -eux
+
+PROGRAM_NAME="${0##*/}"
+# Set defaults
+secret_key_file="../private.pem"
+public_key_file="../public.pem"
+HELP=false
+
+# Read input args
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        -h|--help) HELP=true ;;
+        -d|--directory) content_dir="$2"; shift ;;
+        -s|--secret-key) secret_key_file="$2"; shift ;;
+        -p|--public-key) public_key_file="$2"; shift ;;
+        *) echo "Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+if [ "$HELP" = true ] ; then
+    echo "Usage: $PROGRAM_NAME [options]"
+    echo ""
+    echo "Signs all .json files within input directory"
+    echo ""
+    echo "Options:"
+    echo "-h, --help                Show this help message and exit."
+    echo "-d, --directory <path>    Required. Input folder containing .json files to sign."
+    echo "-s, --secret-key <path>   Required. Path to secret key used for signing."
+    echo "                          Default: '../private.pem'"
```

Comment:
If there is a default, is the arg really required? And if it is the default, shouldn't the parameter reference below be ${secret_key_file:-../private.pem} ? Same for --public-key

Reply:
I meant that it is required because the script will error if there is no key at the specified location. Microsoft has done this in other documentation, example: [DownloadPipelineArtifact task](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/download-pipeline-artifact-v2?view=azure-pipelines)[.](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/download-pipeline-artifact-v2?view=azure-pipelines).) Here, we can see that `buildType` is required, but has a default

Also, I am unsure what your comment is suggesting in the 'parameter reference' section

Reply:
I think `buildType` being required and having a default is confusing and I have submitted feedback on that page stating such. 

No where in the shell script is `../private.pem` used (other than being echoed in the usage). If I ran this script without  `--secret-key`, I would get an unbound variable error. But your usage says it has a default, so how can it be unbound? If I don't give `--secret-key`, shouldn't the script use `../private.pem`? That's what the usage says will happen. So, in my view, this is broken. If `--secret-key` is not given on the command line, then `secret_key_file` should default to `../private.pem'. The bash way to do that is with `${property:-default}` or `${property:=default}`. There is a subtle difference in the two which I encourage you to find out about in the bash man page. Look for details on bash parameter expansion.  

Of course, if your intent is to error with an `unbound variable` if `--secret-key` is missing, then remove the mention of the default (in both places, obviously). 

I'll toss this out there, too. Try this from a bash command line: `${secret_key_file:?--secret-key parameter is required}` and see what happens. Is that feedback more useful than "unbound variable"?

Lastly, there are ways to put error hooks into a bash script. Typically, these are used to show the command line usage. I'm not saying you need to do that here, but it is fun stuff to play with.

Reply:
Thanks for the feedback David! I have a quick question about the error you are getting though: I am unable to reproduce the unbound variable error you mentioned. Here is the result of running the script both with and without the `--secret-key` variable explicitly defined. In both cases, it ran as expected without error (and I believe it would error if there was an unbound variable exception since I have `set -eu` at the top of the script):
![image](https://github.com/microsoft/openjdk-adoptium-marketplace-data/assets/111303274/49a55734-7129-46a0-86b0-40d51fc36b13)

Reply:
OMG. My bad! I totally didn't see 
```
# Set defaults
secret_key_file="../private.pem"
public_key_file="../public.pem"
```
I still stand by my point about "required" and having a default.